### PR TITLE
Cleanup job for GEFS

### LINKED ISSUE #833: Update cleanup job to scrub ensemble data
**Description**
As forecast ensemble jobs are added to the global workflow, we need to ensure the output is being cleaned up properly once it is no longer needed.

**Requirements**
The cleanup job (currently part of archive) needs to properly scrub ensemble data once it is has been archived and is no longer needed. Until ensemble verification is added as part of the workflow instead of a stand-alone system, this may involve a smaller set of data than current deterministic, which does verification within the workflow.

This will be an ongoing issue completed incrementally as pieces of the forecast ensemble are added.

**Acceptance Criteria (Definition of Done)**
A cleanup job that properly removes old ensemble data once it is no longer needed.

**Dependencies**
Add atm-only ensemble free-forecast (#822)
Add atm ensemble product generation (#823)

### PR DESCRIPTION
**Pull request by AntonMFernando-NOAA**
# Description

As forecast ensemble jobs are added to the global workflow, this PR ensures the output is being cleaned up properly once it is no longer needed.
  Resolves #833 
# Type of change

- A cleanup job that properly removes old ensemble data once it is no longer needed.
- Add the cleanup job to the rocoto mesh.

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO 

# How has this been tested?

- CI test on HERCULES
# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: parm/config/gefs/config.cleanup [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# Specify the list of files to exclude from the first stage of cleanup
+# Because arrays cannot be exported, list is a single string of comma-
+# separated values. This string is split to form an array at runtime.
+case ${RUN} in
+	gdas | gfs)	exclude_string="*prepbufr*, *cnvstat*, *atmanl.nc" ;;
+	enkf*)      exclude_string="*f006.ens*" ;;
+	*)			exclude_string="" ;;
+esac
+export exclude_string
```
Comment (aerorahul) [inline parm/config/gefs/config.cleanup]:
```suggestion
export exclude_string=""
```

File: parm/config/gefs/config.cleanup [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+fi
+
+# Specify the list of files to exclude from the first stage of cleanup
+# Because arrays cannot be exported, list is a single string of comma-
+# separated values. This string is split to form an array at runtime.
+case ${RUN} in
+	gdas | gfs)	exclude_string="*prepbufr*, *cnvstat*, *atmanl.nc" ;;
+	enkf*)      exclude_string="*f006.ens*" ;;
+	*)			exclude_string="" ;;
+esac
```
Comment (aerorahul) [inline parm/config/gefs/config.cleanup]: This is `RUN=gefs`, so there is no `case` required here.

**Review by aerorahul — COMMENTED** [review]
looks good.
just a couple of comments.

File: parm/config/gefs/config.cleanup (hunk 2 above)
Comment (WalterKolczynski-NOAA) [reply parm/config/gefs/config.cleanup]: Do we need a separate GEFS config file at all? Or can we just symlink the gfs version as we are doing for several others like config.com?

Comment (aerorahul) [reply parm/config/gefs/config.cleanup]: that would work too.

Comment (AntonMFernando-NOAA) [reply parm/config/gefs/config.cleanup]: added the symlink

File: workflow/rocoto/gefs_tasks.py [hunk 3]
```diff
@@ -564,3 +564,47 @@ def arch(self):
...
+        if self.app_config.do_wave:
+            dep_dict = {'type': 'metatask', 'name': 'wave_post_grid'}
+            deps.append(rocoto.add_dependency(dep_dict))
+            dep_dict = {'type': 'metatask', 'name': 'wave_post_pnt'}
+            deps.append(rocoto.add_dependency(dep_dict))
+            if self.app_config.do_wave_bnd:
+                dep_dict = {'type': 'metatask', 'name': 'wave_post_bndpnt'}
+                deps.append(rocoto.add_dependency(dep_dict))
+                dep_dict = {'type': 'metatask', 'name': 'wave_post_bndpnt_bull'}
+                deps.append(rocoto.add_dependency(dep_dict))
```
Comment (WalterKolczynski-NOAA) [inline workflow/rocoto/gefs_tasks.py]: Archive should already depend on all these things, so I don't think you need them here if you are already depending on that.

Comment (WalterKolczynski-NOAA) [reply workflow/rocoto/gefs_tasks.py]: Incidentally, I think the archive job needs to add a dependency on `extractvar` if it is being run.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
We will run this through the normal CI tests when it is complete, but to test it you'll need to run several cycles to test the cleanup bit properly. We don't keep a bunch of GEFS initial conditions around, so work with the ensemble team to set that up.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending CI and manual completion of a longer GEFS test.

**Review by aerorahul — APPROVED** [review]

**Review by aerorahul — APPROVED** [review]

**Pull request merged**